apis/addonconfigs/cni/v1alpha2/antreaconfig_types.go

```diff
@@ -288,51 +288,166 @@ type AntreaNsx struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default:=false
 	Enable bool `json:"enable,omitempty"`
-	// BootstrapFrom either providerRef or inline configs
-	// +kubebuilder:validation:Optional
-	BootstrapFrom AntreaNsxBootstrapFrom `json:"bootstrapFrom,omitempty"`
 	// Config is  configuration for nsxt adapter
 	// +kubebuilder:validation:Optional
 	AntreaNsxConfig AntreaNsxConfig `json:"config,omitempty"`
 }
 
-type AntreaNsxBootstrapFrom struct {
-	// ProviderRef is used with uTKG, which will be filled by uTKG Addon Controller
+type AntreaNsxConfig struct {
+	// BootstrapFrom either providerRef or inline configs
 	// +kubebuilder:validation:Optional
-	ProviderRef *AntreaNsxProvider `json:"providerRef,omitempty"`
-	// Inline is used with TKGm, user need to fill in manually
+	BootstrapFrom string `json:"bootstrapFrom,omitempty"`
+
+	// InfraType is the type for infrastructure, so far it is vSphere, VMC, AWS, Azure
 	// +kubebuilder:validation:Optional
-	Inline *AntreaNsxInline `json:"inline,omitempty"`
-}
+	InfraType string `json:"infraType,omitempty"`
 
-type AntreaNsxProvider struct {
-	// Api version for nsxServiceAccount, its value is "nsx.vmware.com/v1alpha1" now
 	// +kubebuilder:validation:Optional
-	ApiGroup string `json:"apigroup,omitempty"`
-	// Kind is the kind for crd, here its value is NsxServiceAccount
+	NSXCert string `json:"nsxCert,omitempty"`
```

Comment:
I see that antreaconfig_util.go uses a getNSXCert function to get tls.crt and tls.key content from a Secret. I think that we can remove the NSXCert and NSXKey from the AntreaConfig CRD. Instead, we can introduce a NSXSecretName string field. It points to a Secret resource by name in the same Namespace. Then antreaconfig_util.go can refers to the NSXSecretName and pass it to getNSXCert function.

Reply:
done.